workers/unity/Packages/io.improbable.gdk.tools/GenerateCode.cs

```diff
@@ -211,13 +255,44 @@ private static void ProcessStdOut(string output)
                         break;
                 }
             }
-            else
+        }
+
+        private static void ProcessCodegenOutput(string output)
+        {
+            try
+            {
+                var log = CodegenLog.FromRaw(output);
+                latestCodegenLogs.Append(log);
+
+                switch (log.Level)
```

Comment:
If you write a `LogType GetUnityLogType()` method on the `CodegenLog` struct then this entire switch/case can just be:

```
Debug.logger.Log(log.GetUnityLogType(), log.Logger, log.Message);
```